pyproject.toml

```diff
@@ -95,10 +95,10 @@ docs = [
   "nc-time-axis >=1.4.1",
   "pooch >=1.8.0",
   "pygments",
-  "sphinx >=8.2.2",
+  "sphinx >=7.1.0,<8.2.0", # pinned until nbsphinx supports sphinx 8.2
   "sphinx-autobuild >=2024.4.16",
-  "sphinx-autodoc-typehints >=3.1.0",
-  "sphinx-codeautolink >=0.16.2",
```

Comment:
I think keeping this pin is fine, but good call on the others. Sphinx 8.2 dropped python 3.10 support.